File: packages/python-sdk/src/blindfold/regex/detectors/hu/tax_id.py
"""Hungarian Tax ID detector"""

import re

from ...base import RegexDetector
from ...entities import EntityType
from ...registry import register_region
from ...validators import hu_tax_id_checksum


@register_region("hu")
class TaxIdDetector(RegexDetector):
    entity_type = EntityType.HU_TAX_ID
    score = 0.85
    context_keywords = [
        "adoazonosito", "adószám", "tax id", "tax number",
        "adoazonosito jel",
    ]
    context_required = True
    context_window = 50

    # Starts with 8, followed by 9 digits
    pattern = re.compile(
        r"\b8\d{9}\b"
    )
    validator = staticmethod(hu_tax_id_checksum)

File: packages/js-sdk/src/regex/detectors/hu/tax-id.ts
// Hungarian Tax ID detector

import { RegexDetector } from '../../base'
import { EntityType } from '../../entities'
import { registerRegion } from '../../registry'
import { huTaxIdChecksum } from '../../validators'

class HuTaxIdDetector extends RegexDetector {
  entityType = EntityType.HU_TAX_ID
  score = 0.85
  contextKeywords = [
    'adoazonosito',
    'adószám',
    'tax id',
    'tax number',
    'adoazonosito jel',
  ]
  contextRequired = true
  contextWindow = 50

  pattern = /\b8\d{9}\b/g
  validator = huTaxIdChecksum
}

registerRegion('hu', HuTaxIdDetector)

export { HuTaxIdDetector }
